src/utils/upload/index.ts

```diff
@@ -120,3 +131,60 @@ export function areFileWithCollectionsSame(
 ): boolean {
     return firstFile.localID === secondFile.localID;
 }
+
+export function analyseUploadFiles(
+    uploadType: UPLOAD_TYPE,
+    toUploadFiles: File[] | ElectronFile[]
+): AnalysisResult {
+    if (isElectron() && uploadType === UPLOAD_TYPE.FILES) {
+        return NULL_ANALYSIS_RESULT;
+    }
+
+    const paths: string[] = toUploadFiles.map((file) => file['path']);
+    const getCharCount = (str: string) => (str.match(/\//g) ?? []).length;
+    paths.sort((path1, path2) => getCharCount(path1) - getCharCount(path2));
+    const firstPath = paths[0];
+    const lastPath = paths[paths.length - 1];
+
+    const L = firstPath.length;
+    let i = 0;
+    const firstFileFolder = firstPath.substring(0, firstPath.lastIndexOf('/'));
+    const lastFileFolder = lastPath.substring(0, lastPath.lastIndexOf('/'));
+    while (i < L && firstPath.charAt(i) === lastPath.charAt(i)) i++;
+    let commonPathPrefix = firstPath.substring(0, i);
+
+    if (commonPathPrefix) {
+        commonPathPrefix = commonPathPrefix.substring(
+            0,
+            commonPathPrefix.lastIndexOf('/')
+        );
+        if (commonPathPrefix) {
+            commonPathPrefix = commonPathPrefix.substring(
+                commonPathPrefix.lastIndexOf('/') + 1
+            );
+        }
+    }
+    return {
+        suggestedCollectionName: commonPathPrefix || null,
+        multipleFolders: firstFileFolder !== lastFileFolder,
+    };
+}
+export function getCollectionWiseFiles(toUploadFiles: File[] | ElectronFile[]) {
```

Comment:
rename to getCollectionToFilesMap or some thing similar? Also, if not a big change, let's define return type too?

Edit: Before reading the code, I thought it was mapping collectionID to files. But that's definitely not the case. 
Let's rename it so that it's easy to guess what the method is doing without reading the code.